lib/translate/inquiry.cc

```diff
@@ -0,0 +1,308 @@
+      for (int i = 4; i < result.page_length; i += 5) {
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(result.product_serial_number, formatted_hex_string,
+             result.page_length);
+    } else {
+      // 6.1.3.1.2
+      result.page_length = 20;
+
+      // convert 128-bit number into hex string
+      char hex_string[17];
+      sprintf(hex_string, "%08llx", identify_namespace_data.eui64);
+
+      // insert _ and . in the correct positions
+      char formatted_hex_string[21];
+      for (int i = 4; i < result.page_length; i += 5) {
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(result.product_serial_number, formatted_hex_string,
+             result.page_length);
+    }
+  } else {
+    if (nguid_nz) {
+      // 6.1.3.1.1
+      result.page_length = 40;
+
+      // convert 128-bit number into hex string
+      char hex_string[33];
+      sprintf(hex_string, "%08llx", identify_namespace_data.nguid[0]);
+      sprintf(&hex_string[16], "%08llx", identify_namespace_data.nguid[1]);
+
+      // insert _ and . in the correct positions
+      char formatted_hex_string[41];
+      for (int i = 4; i < result.page_length; i += 5) {
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(result.product_serial_number, formatted_hex_string,
+             result.page_length);
+    } else {
+      // 6.1.3.1.3
+      // valid for NVMe 1.0 devices only
+      // TODO?
+    }
+  }
+
+  return result;
+}
+
+scsi_defs::UnitSerialNumber BuildUnitSerialNumberVpd() {
+  nvme_defs::IdentifyNamespace identify_namespace_data =
+      NvmeIdentifyNamespace();
+  return TranslateUnitSerialNumberVpdResponse(identify_namespace_data);
+}
+
+// TODO: write return value to a buffer
+// Main logic engine for the Inquiry command
+void translate(absl::Span<const uint32_t> raw_cmd) {
+  std::optional<scsi_defs::InquiryCommand> opt_cmd = RawToScsiCommand(raw_cmd);
+  if (!opt_cmd.has_value()) return;
+
+  scsi_defs::InquiryCommand cmd = opt_cmd.value();
+  if (cmd.evpd) {
+    switch (cmd.page_code) {
+      // Shall be supported by returning Supported Vpd Pages data page to
```

Comment:
`Shall be supported` implies it is not currently supported/ implemented, however this case (and below) clearly has a call to `BuildSupportedVpdPages`. It feels as if the comment is contradicting the code. In general, this comment is a bit redundant since the name of the function below tells us the same information. I would reduce this comment to keep only the reference to 6.1.2, and specify what 6.1.2 is referencing (I'm guessing the scsi interface manual?)

Also, move the comment to inside its case, to make it obvious what the comment refers to